Update user guide with new features (#178)
AY2021S2-CS2103-W16-3/tp: docs/UserGuide.md

### Adding contacts : `add`

Adds a person to PartyPlanet's Contacts List.

Format: `add -n NAME [-p PHONE_NUM] [-e EMAIL] [-a ADDRESS] [-t TAG]…​ [-b BIRTHDAY]​ [-r REMARK]​`<br>

> * The birthday must be in a valid date format, with or without year, and must be in the past, e.g. 13 Jan

Reviewer: ```suggestion
* The birthday must be in a valid date format, with or without year, and must be in the past if the year is specified, e.g. 13 Jan, 13 Mar 1997
```

* `tags` lists out all tags available.
* `tags -f cs2103` lists out all tags that contain `cs2103`.

### Undoing actions : `undo`

Undoes the most recent action that changed PartyPlanet's Contact List.

> Can be invoked repeated until there is no more history from that session.

Reviewer: ```suggestion
Can be invoked repeatedly until there is no more history from the current session.
```

Format: `undo`

### Adding events : `eadd`

Adds an event to PartyPlanet's Events list. Similar to `add` for person contacts.

Format: `eadd -n NAME [-d DATE] [-r REMARK]`

> * The date must be in a valid date format with year, e.g. 2022-05-07

Reviewer: ```suggestion
* The date must be in a valid date format with year, e.g. 2022-05-07, 2 feb 2021
```
